src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java

```diff
@@ -0,0 +1,448 @@
+			json.put("jwkFilePath", jwkFilePath);
+			json.put("trustMarksFilePath", trustMarksFilePath);
+
+			return json;
+		}
+
+		public String getEncrJwkFilePath() {
+			return encrJwkFilePath;
+		}
+
+		public void setEncrJwkFilePath(String encrJwkFilePath) {
+			this.encrJwkFilePath = encrJwkFilePath;
+		}
+
+		private String applicationName;
+		private String applicationType;
+		private Set<String> contacts = new HashSet<>();
+		private Set<String> scope = new HashSet<>();
+		private String clientId;
+		private Set<String> redirectUris = new HashSet<>();
+		private Set<String> requestUris = new HashSet<>();
+		private String jwkFilePath;
+		private String trustMarksFilePath;
+		private String encrJwkFilePath;
+
+	}
+
+	public static class OpenIdCredentialIssuer {
+		private String credentialIssuer;
+		private String id;
+		private String authorizationEndpoint;
+		private String tokenEndpoint;
+		private String pushedAuthorizationRequestEndpoint;
+		private String credentialEndpoint;
+		private Set<String> dpopSigningAlgValuesSupported = new HashSet<>();
+		private String jwkFilePath;
+		private String mdocJwkFilePath;
+		private String encrJwkFilePath;
+		private String sub;
+		private List<String> trustChain = new ArrayList<>();
+
+		public String getId() {
+			return id;
+		}
+
+		public void setId(String id) {
+			this.id = id;
+		}
+
+		public String getCredentialIssuer() {
+			return credentialIssuer;
+		}
+
+		public void setCredentialIssuer(String credentialIssuer) {
+			this.credentialIssuer = credentialIssuer;
+		}
+
+		public String getAuthorizationEndpoint() {
+			return authorizationEndpoint;
+		}
+
+		public void setAuthorizationEndpoint(String authorizationEndpoint) {
+			this.authorizationEndpoint = authorizationEndpoint;
+		}
+
+		public String getTokenEndpoint() {
+			return tokenEndpoint;
+		}
+
+		public void setTokenEndpoint(String tokenEndpoint) {
+			this.tokenEndpoint = tokenEndpoint;
+		}
+
+		public String getPushedAuthorizationRequestEndpoint() {
+			return pushedAuthorizationRequestEndpoint;
+		}
+
+		public void setPushedAuthorizationRequestEndpoint(String pushedAuthorizationRequestEndpoint) {
+			this.pushedAuthorizationRequestEndpoint = pushedAuthorizationRequestEndpoint;
+		}
+
+		public String getCredentialEndpoint() {
+			return credentialEndpoint;
+		}
+
+		public void setCredentialEndpoint(String credentialEndpoint) {
+			this.credentialEndpoint = credentialEndpoint;
+		}
+
+		public Set<String> getDpopSigningAlgValuesSupported() {
+			return Collections.unmodifiableSet(dpopSigningAlgValuesSupported);
+		}
+
+		public void setDpopSigningAlgValuesSupported(Set<String> dpopSigningAlgValuesSupported) {
+			this.dpopSigningAlgValuesSupported = dpopSigningAlgValuesSupported;
+		}
+
+		public String getJwkFilePath() {
+			return jwkFilePath;
+		}
+
+		public void setJwkFilePath(String jwkFilePath) {
+			this.jwkFilePath = jwkFilePath;
+		}
+
+		public String getMdocJwkFilePath() {
+			return mdocJwkFilePath;
```

Comment:
why we have mdoc here?

this is out of the scope of openid federation

Reply:
Removed.